### PR TITLE
[SPARK-42216][CORE][TESTS] Fix two check conditions and remove redundant `toMap` in `util.JsonProtocolSuite`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The main changes of this pr as follows:
- Remove explicit `import scala.collection.Map` and use the default `immutable.Map` in `util.JsonProtocolSuite`,  and  also clean up the redundant `toMap` conversion after remove explicit `import scala.collection.Map`

- Fix two `assert` conditions: change `assert(e1.x === e1.x)` to `assert(e1.x === e2.x)`, the originally assert is always true.

### Why are the changes needed?
- Use `immutable.Map` by default in `util.JsonProtocolSuite` and remove redundant `toMap` conversion for code cleanup
- fix two wrong assert conditions in `util.JsonProtocolSuite`

### Does this PR introduce _any_ user-facing change?
No, just for test

### How was this patch tested?

- Pass GitHub Actions
- Manual test with Scala 2.13
```
gh pr checkout 39772
dev/change-scala-version.sh 2.13 
mvn clean install -pl core -am -Pscala-2.13 -Dtest=none -DwildcardSuites=org.apache.spark.util.JsonProtocolSuite
```

```
Run completed in 4 seconds, 319 milliseconds.
Total number of tests run: 34
Suites: completed 2, aborted 0
Tests: succeeded 34, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
```
all test passed

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala [hunk 1]
```diff
@@ -924,12 +923,12 @@ private[spark] object JsonProtocolSuite extends Assertions {
...
       case (e1: SparkListenerEnvironmentUpdate, e2: SparkListenerEnvironmentUpdate) =>
-        assertEquals(e1.environmentDetails, e2.environmentDetails)
+        assertEquals(e1.environmentDetails.toMap, e2.environmentDetails.toMap)
       case (e1: SparkListenerExecutorAdded, e2: SparkListenerExecutorAdded) =>
-        assert(e1.executorId === e1.executorId)
+        assert(e1.executorId === e2.executorId)
         assertEquals(e1.executorInfo, e2.executorInfo)
       case (e1: SparkListenerExecutorRemoved, e2: SparkListenerExecutorRemoved) =>
-        assert(e1.executorId === e1.executorId)
+        assert(e1.executorId === e2.executorId)
```
Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala:931]: According to the PR description, after fixing this, `toMap` causes failure, right?

**Review by dongjoon-hyun — COMMENTED** [review]
Since this is related to `Map`, please test Scala 2.13 together.

**Comment by LuciferYang** [comment]
> Since this is related to `Map`, please test Scala 2.13 together.

done

File: core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala (hunk 1 above)
Comment (LuciferYang) [reply core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala:931]: No, two things have been done in this pr : one is code cleanup, the other is bug fix.

Is it better to divide this pr into two parts?

Sorry, my descriptive ability is poor. It seems that there is a misunderstanding ...

Comment (LuciferYang) [reply core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala:931]: > According to the PR description, after fixing this, `toMap` causes failure, right?

After fixing this, `toMap` is redundant collection conversion

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs)

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Pull request closed without merging**